src/spgwu/simpleswitch/spgwu_s1u.cpp

```diff
@@ -116,6 +116,9 @@ void spgwu_s1u::handle_receive(
     char* recv_buffer, const std::size_t bytes_transferred,
     const endpoint& r_endpoint) {
 #define GTPU_MESSAGE_FLAGS_POS_IN_UDP_PAYLOAD 0
+#define GTPU_MESSAGE_EXT_HEADER_FLAG 0x34
+#define GTPU_MESSAGE_SEQN_FLAG 0x36
+#define GTPU_MESSAGE_PRIME_FLAG 0x07
```

Comment:
These constants could be set in gtpv1u/gtpu.h
Ext header bit for example is not 0x34 but 0x4, you optimize the code, but the compiler will do it for you

Reply:
HI Lionel, What  0x34 should be defined ?

Reply:
Because it is combination of type, version and ext_hdr_present flag.

Reply:
You could do this:
#define GTP_VERSION_1 1
#define GTPU_MESSAGE_SN_MASK 0x2
#define GTPU_MESSAGE_EXT_HEADER_MASK 0x4
#define GTPU_MESSAGE_PT_MASK 0x10
#define GTPU_MESSAGE_VERSION_MASK 0xE
(in gtpu.h)
then 
if (
     (
       (gtp_flags & GTPU_MESSAGE_VERSION_MASK) == GTP_VERSION_1) && (gtp_flags & GTPU_MESSAGE_PT_MASK)
     ) 
    && 
     (
        (gtp_flags & GTPU_MESSAGE_EXT_HEADER_MASK) || (gtp_flags & GTPU_MESSAGE_SN_MASK)
     )
   ) {
...
}

Reply:
@lionelgo 
Looks okay now ?

Reply:
One more thing : we should also test the PN flag (forgot to mention previously)